user_modules/script_runner.lua

```diff
@@ -61,24 +93,55 @@ local function extendedAddTestStep(testStepName, testStepImplFunction, paramsTab
   addTestStep(testStepName, newTestStepImplFunction)
 end
 
---[[ Title + Step approach]]
-local function buildTitle(titleText)
-  local maxLength = 101
-  local filler = "-"
-  local resultTable = {}
-  for line in titleText:gmatch("[^\n]+") do
-    local lineLength = #line
-    if lineLength >= maxLength then
-      table.insert(resultTable, line)
-    else
-      local tailLength = math.fmod(maxLength - lineLength, 2)
-      local emtyLineSideLength = math.floor((maxLength - lineLength) / 2)
-      table.insert(resultTable, filler:rep(emtyLineSideLength) .. line .. filler:rep(emtyLineSideLength + tailLength))
+local function isTestApplicable(testApplicableSdlSettings)
+  if next(testApplicableSdlSettings) == nil then
+   return true
+  end
+  local isMatched = true
+  for _, sdlSettingsSet in pairs(testApplicableSdlSettings) do
+    for sdlSettingsItem, listOfValuesForItem in pairs(sdlSettingsSet) do
+      local sdlBuildOptionValue = Test.sdlBuildOptions[sdlSettingsItem]
+      if sdlBuildOptionValue then
+        if not existsInList(listOfValuesForItem, sdlBuildOptionValue) then
+          isMatched = false
+          break
+        end
+      else
+        return error("SDL build option " .. sdlSettingsItem .. " does not specified!")
+      end
+    end
+    if isMatched then
+      return true
     end
+    isMatched = true
   end
-  return table.concat(resultTable, "\n")
+  return false
+end
+
+local function skipTest(reason)
+  title = ""
```

Comment:
@aderiabin "title" is not used

Reply:
@dboltovskyi Will not be fixed.

This line is used for clear module variable `title` before build skip title using `runner.Title("TEST SKIPPED")` call. 
In other words: This line is to skip print of all titles for test in case test is skipped.
(please look at example below)

Example:
Test script:
```
...
runner.Title("Preconditions")
runner.Step("Clean environment", commonRC.preconditions)
...
```
1) script_runner with `title = ""`:
```
...
local function skipTest(reason)
  title = ""
  runner.Title("TEST SKIPPED")
...
```
Result in console:
```
...
Test severity: Critical
=====================================================================================================
--------------------------------------------TEST SKIPPED---------------------------------------------
Test is incompatible with current build configuration of SDL:
	{
		extendedPolicy = "HTTP",
		remoteControl = "ON"
	}
...
```
2) script_runner without `title = ""`:
```
...
local function skipTest(reason)
  runner.Title("TEST SKIPPED")
...
```
Result in console:
```
...
Test severity: Critical
=====================================================================================================
--------------------------------------------Preconditions--------------------------------------------
--------------------------------------------TEST SKIPPED---------------------------------------------
Test is incompatible with current build configuration of SDL:
	{
		remoteControl = "ON",
		extendedPolicy = "HTTP"
	}
...
```